qiskit/primitives/sampler.py

```diff
@@ -0,0 +1,107 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Sampler class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info import Statevector
+from qiskit.result import QuasiDistribution
+
+from .base_sampler import BaseSampler
+from .sampler_result import SamplerResult
+from .utils import final_measurement_mapping, init_circuit
+
+
+class Sampler(BaseSampler):
+    """
+    Sampler class
+    """
+
+    def __init__(
+        self,
+        circuits: QuantumCircuit | Iterable[QuantumCircuit],
+        parameters: Iterable[Iterable[Parameter]] | None = None,
+    ):
+        """
+        Args:
+            circuits: circuits to be executed
+
+        Raises:
+            QiskitError: if some classical bits are not used for measurements.
+        """
+        if isinstance(circuits, QuantumCircuit):
+            circuits = [circuits]
+        circuits = [init_circuit(circuit) for circuit in circuits]
+        q_c_mappings = [final_measurement_mapping(circuit) for circuit in circuits]
+        self._qargs_list = []
+        for circuit, q_c_mapping in zip(circuits, q_c_mappings):
+            if set(range(circuit.num_clbits)) != set(q_c_mapping.values()):
+                raise QiskitError(
+                    "some classical bits are not used for measurements."
+                    f" the number of classical bits {circuit.num_clbits},"
+                    f" the used classical bits {set(q_c_mapping.values())}."
```

Comment:
Yes. You are right. It is possible to extend it that way, but we did not implement it because it would complicate the code to add zeros in the post-processing. If it is necessary, we would extend it. (but it is not easy, so let me make it the next PR.)